raphtory-graphql/src/model/mod.rs

```diff
@@ -143,6 +146,63 @@ impl Mut {
         keys
     }
 
+    async fn rename_graph<'a>(
+        ctx: &Context<'a>,
+        parent_graph_name: String,
+        graph_name: String,
+        new_graph_name: String,
+    ) -> Result<bool> {
+        if new_graph_name.ne(&graph_name) && parent_graph_name.ne(&graph_name) {
+            let mut data = ctx.data_unchecked::<Data>().graphs.write();
+
+            let subgraph = data.get(&graph_name).ok_or("Graph not found")?;
+            let path = subgraph
+                .static_prop(&"path".to_string())
+                .expect("Path is missing")
+                .to_string();
+
+            let parent_graph = data.get(&parent_graph_name).ok_or("Graph not found")?;
+            let new_subgraph = parent_graph
+                .subgraph(subgraph.vertices().iter().map(|v| v.name()).collect_vec())
+                .materialize()
+                .expect("Failed to materialize graph");
+
+            let static_props_without_name: Vec<(String, Prop)> = subgraph
+                .properties()
+                .into_iter()
+                .filter(|(a, b)| a != "name")
+                .collect_vec();
+
+            new_subgraph
+                .add_constant_properties(static_props_without_name)
+                .expect("Failed to add static properties");
```

Comment:
I don't think we want a load of expects in here. We shouldn't get an issue, but best to handle it